crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -680,16 +680,91 @@ impl Analyzer<'_, '_> {
             return never!();
         }
 
+        if types.into_iter().filter(|t| t.is_enum_variant()).count() >= 2 {
```

Comment:
This is wrong because two enum variants can have a same value

Reply:
```ts
enum Tag{
  a = 1,
  b = 1,
}

declare const s1: Tag.a & Tag.b;

```

<img width="439" alt="image" src="https://user-images.githubusercontent.com/29931815/202369023-75066f0c-0dec-4f76-9eee-b21055698f27.png">

Reply:
um....

i understand and will add test code
thx.

Postscript.
But who writes code like that...

Reply:
Working on compiler is annoying because we have to handle useless codes 🤣